### ISSUE TITLE
riak_core_bucket_type:get/1 handles default type improperly

### ISSUE THREAD

**Issue by jrwest** [basho/riak_core#496]
Reported by @alexmoore. Found via http interface which assumes `default` exists, as it should. The correct thing to do would be return the default type properties from the `riak_core_bucket_props:defaults/0`.

``````
curl http://localhost:10018/types/default/props                                                                                                                                                                                                              dev/dev1 | master
<html><head><title>500 Internal Server Error</title></head><body><h1>Internal Server Error</h1>The server encountered an error while processing this request:<br><pre>{error,
    {error,function_clause,
        [{riak_kv_wm_bucket_type,'-produce_bucket_type_body/2-lc$^0/1-0-',
             [undefined],
             [{file,"src/riak_kv_wm_bucket_type.erl"},{line,244}]},
         {riak_kv_wm_bucket_type,produce_bucket_type_body,2,
             [{file,"src/riak_kv_wm_bucket_type.erl"},{line,244}]},
         {webmachine_resource,resource_call,3,
             [{file,"src/webmachine_resource.erl"},{line,186}]},
         {webmachine_resource,do,3,
             [{file,"src/webmachine_resource.erl"},{line,142}]},
         {webmachine_decision_core,resource_call,1,
             [{file,"src/webmachine_decision_core.erl"},{line,48}]},
         {webmachine_decision_core,decision,1,
             [{file,"src/webmachine_decision_core.erl"},{line,558}]},
         {webmachine_decision_core,handle_request,2,
             [{file,"src/webmachine_decision_core.erl"},{line,33}]},
         {webmachine_mochiweb,loop,2,
             [{file,"src/webmachine_mochiweb.erl"},{line,74}]}]}}</pre><P><HR><ADDRESS>mochiweb+webmachine web server</ADDRESS></body></html>%                                                                                                                                   ```
``````

**Comment by jrwest** [comment]
fixed w/ merge of #499

**Comment by lukebakken** [comment]
Verified fixed in pre10:

```
[root@FREEBSD-91-YZTEST-0 ~]# curl 'http://localhost:8098/types/default/props'
{"props":{"allow_mult":true,"basic_quorum":false,"big_vclock":50,"chash_keyfun":{"mod":"riak_core_util","fun":"chash_std_keyfun"},"dw":"quorum","last_write_wins":false,"linkfun":{"mod":"riak_kv_wm_link_walker","fun":"mapreduce_linkfun"},"n_val":3,"notfound_ok":true,"old_vclock":86400,"postcommit":[],"pr":0,"precommit":[],"pw":0,"r":"quorum","rw":"quorum","small_vclock":50,"w":"quorum","young_vclock":20}}
```
